File: problems/problems_3047/solution.ts
function largestSquareArea(bottomLeft: number[][], topRight: number[][]): number {
    
};

export function Solve(inputJsonElement: string): any {
	const inputValues: string[] = inputJsonElement.split("\n");
	const bottomLeft: number[][] = JSON.parse(inputValues[0]);
	const topRight: number[][] = JSON.parse(inputValues[1]);
	return largestSquareArea(bottomLeft, topRight);
}

File: problems/problems_3047/solution.rs
use serde_json::{json, Value};

pub struct Solution;

impl Solution {
    pub fn largest_square_area(bottom_left: Vec<Vec<i32>>, top_right: Vec<Vec<i32>>) -> i64 {
        
    }
}

#[cfg(feature = "solution_3047")]
pub fn solve(input_string: String) -> Value {
	let input_values: Vec<String> = input_string.split('\n').map(|x| x.to_string()).collect();
	let bottom_left: Vec<Vec<i32>> = serde_json::from_str(&input_values[0]).expect("Failed to parse input");
	let top_right: Vec<Vec<i32>> = serde_json::from_str(&input_values[1]).expect("Failed to parse input");
	json!(Solution::largest_square_area(bottom_left, top_right))
}
